File: backend/Core/Persons/Person.cs
namespace EquiPressure.Core.Persons;

public class Person
{
    public int PersonId { get; set; }

    public required string Firstname { get; set; }
    public required string Lastname { get; set; }
    public required string Email { get; set; }

    public int HeightCm { get; set; }
    public int WeightKg { get; set; }

    public int Plz { get; set; }
    public required string City { get; set; }
    public required string Street { get; set; }
    public int Housenumber { get; set; }

    public string? WebsiteLink { get; set; }
    public string? Description { get; set; }
    
    public List<CustomerRelationship> CustomerRelationships { get; set; } = [];
    public List<Role> Roles { get; set; } = [];
}

// Assoziationstabelle 
public class CustomerRelationship
{
    public int PersonId { get; set; }
    public int RelatedPersonId { get; set; }

    // erscheint in der Favouriten-Liste unserer Person
    public bool IsFavorite { get; set; }

    // erscheint in der Kontakt-Liste unserer Person
    public bool IsContact { get; set; }
}

public class Role
{
    public int RoleId { get; set; }
    public required string Name { get; set; }
}

File: reiter_frontend/lib/models/person.dart
class Person {
  final int id;

  final String firstname;
  final String lastname;

  final int heightCm;
  final int weightKg;

  final int plz;
  final String city;
  final String street;
  final int housenumber;

  final String email;
  final String? websiteLink;
  final String? description;
  final DateTime? dob;

  final bool hidden;

  const Person({
    required this.id,
    required this.firstname,
    required this.lastname,
    required this.heightCm,
    required this.weightKg,
    required this.plz,
    required this.city,
    required this.street,
    required this.housenumber,
    required this.email,
    this.websiteLink,
    this.description,
    this.dob,
    this.hidden = false,
  });

  factory Person.fromJson(Map<String, dynamic> json) {
    return Person(
      id: json['person_id'],
      firstname: json['firstname'],
      lastname: json['lastname'],
      heightCm: json['height_cm'],
      weightKg: json['weight_kg'],
      plz: json['plz'],
      city: json['city'],
      street: json['street'],
      housenumber: json['housenumber'],
      email: json['email'],
      websiteLink: json['website_link'],
      description: json['description'],
      dob: json['dob'] != null ? DateTime.parse(json['dob']) : null,
      hidden: json['hidden'] ?? false,
    );
  }

  Map<String, dynamic> toJson() => {
        'person_id': id,
        'firstname': firstname,
        'lastname': lastname,
        'height_cm': heightCm,
        'weight_kg': weightKg,
        'plz': plz,
        'city': city,
        'street': street,
        'housenumber': housenumber,
        'email': email,
        'website_link': websiteLink,
        'description': description,
        'dob': dob?.toIso8601String(),
        'hidden': hidden,
      };

  String get fullName => '$firstname $lastname';
}
